OurUmbraco/Auth/AppBuilderExtensions.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        public static void ConfigureHmacBearerTokenAuthentication(this IAppBuilder appBuilder, string[] authenticatedPaths)
+        {
+            appBuilder.UseOAuthBearerAuthentication(new OAuthBearerAuthenticationOptions
+            {
+                Provider = new OAuthBearerAuthenticationProvider
+                {
+                    //This is the first callback made when OWIN starts
+                    //to process the request, here we need to set the token
+                    //to null if we don't want it to be processed - basically
+                    //this is where we need to:
+                    // * check the current request URL to see if we should auth the request (only deploy end points)
+                    OnRequestToken = context =>
+                    {
+                        var umbPath = GlobalSettings.UmbracoMvcArea;
+
+                        var applicationBasePath = (context.Request.PathBase.HasValue ? context.Request.PathBase.Value : string.Empty).EnsureEndsWith('/');
+                        
+                        var requestPath = context.Request.Uri.CleanPathAndQuery();
+
+                        //Only authenticated endpoints to be authenticated and these must have the projectid
+                        //and memberid headers in the request
+                        if (authenticatedPaths.Any(s => requestPath.StartsWith($"{applicationBasePath}{umbPath}/{s}", StringComparison.InvariantCultureIgnoreCase))
+                            && EnsureHeaderValues(context, ProjectAuthConstants.ProjectIdHeader, ProjectAuthConstants.MemberIdHeader))
+                        {                            
+                            return Task.FromResult(0);                         
+                        }   
+
+                        context.Token = null;
+                        return Task.FromResult(0);
+                    }
+                },
+                AccessTokenProvider = new AuthenticationTokenProvider
+                {
+                    //Callback used to parse the token in the request,
+                    //if the token parses correctly then we should assign a ticket
+                    //to the request, this is the "User" that will get assigned to
+                    //the request with Claims.
+                    //If the token is invalid, then don't assign a ticket and OWIN
+                    //will take care of the rest (not authenticated)
+                    OnReceive = context =>
+                    {
+                        var requestPath = context.Request.Uri.CleanPathAndQuery();
+                        if (!TryGetHeaderValue(context, ProjectAuthConstants.ProjectIdHeader, out var projectId))
+                            throw new InvalidOperationException("No project Id found in request"); // this will never happen
+                        if (!TryGetHeaderValue(context, ProjectAuthConstants.MemberIdHeader, out var memberId))
```

Comment:
You mean no member Id found.. copy/paste? 😉

Reply:
Fixed in https://github.com/umbraco/OurUmbraco/pull/552/commits/6de9aa156112210bd4d89f1e2f6d3450ec3cf6a6